src/Runner/Runner.php

```diff
@@ -63,12 +83,18 @@ final class Runner
 
     private bool $stopOnViolation;
 
+    private ParallelConfig $parallelConfig;
+
+    private ?InputInterface $input;
+
+    private ?string $configFile;
+
     /**
-     * @param \Traversable<\SplFileInfo> $finder
-     * @param list<FixerInterface>       $fixers
+     * @param null|\Traversable<array-key, \SplFileInfo> $fileIterator
+     * @param list<FixerInterface>                       $fixers
      */
     public function __construct(
-        \Traversable $finder,
+        ?\Traversable $fileIterator,
```

Comment:
We have optional iterator, then we have one-item fixInfo array or one-event in ->events in Worker instance.

Consider to externalise `Runner::fixFile()` to a service with method like `::fixFile` that is not handling arrays, events, etc.

That way you won't need to have those changes like optional iterator or setIterator method.

Reply:
I planned to do it this way at the beginning, that's why I extracted `fixFile()` but in the end I decided to not extract it further. It would require passing almost all dependencies of Runner to a new place IIRC. Also, I really don't want to refactor it further in this PR, since it's more like a cosmetics than actually needed change.

Reply:
👍🏻